src/metabase/db/metadata_queries.clj

```diff
@@ -92,14 +92,24 @@
 (s/defn table-rows-sample :- (s/maybe si/TableSample)
   "Run a basic MBQL query to fetch a sample of rows belonging to a Table."
   {:style/indent 1}
-  [table :- si/TableInstance, fields :- [si/FieldInstance]]
-  (let [results ((resolve 'metabase.query-processor/process-query)
-                 {:database   (:db_id table)
-                  :type       :query
-                  :query      {:source-table (u/get-id table)
-                               :fields       (vec (for [field fields]
-                                                    [:field-id (u/get-id field)]))
-                               :limit        max-sample-rows}
-                  :middleware {:format-rows?           false
-                               :skip-results-metadata? true}})]
-    (get-in results [:data :rows])))
+  ([table :- si/TableInstance, fields :- [si/FieldInstance]]
+   (table-rows-sample table fields nil))
+  ([table :- si/TableInstance, fields :- [si/FieldInstance] truncation-size :- (s/maybe s/Int)]
```

Comment:
I think having an optional options map for things such as `truncation-size` is more future-proof, as I can easily see doing things such as different sampling strategies in the future, which will also need to be signalled